src/inmanta/server/services/resourceservice.py

```diff
@@ -1038,3 +1039,62 @@ async def versioned_resource_details(
         if not resource:
             raise NotFound("The resource with the given id does not exist")
         return resource
+
+    @handle(methods_v2.unmanaged_resource_create)
+    async def unmanaged_resource_create(
+        self, env: data.Environment, unmanaged_resource_id: str, values: Dict[str, str]
+    ) -> None:
+        unmanaged_resource = UnmanagedResource(unmanaged_resource_id=unmanaged_resource_id, values=values)
```

Comment:
Why not have the `UnmanagedResource` as a parameter to the API endpoint, instead of passing in the three individual parameters? We do the same in the batch API call.

Reply:
I though for the single calls we wanted to have the api call as close as possible to this one:
```
@method(
    path="/resource/<id>",
    operation="GET",
    agent_server=True,
    validate_sid=False,
    arg_options=ENV_OPTS,
    api=True,
    client_types=[const.ClientType.api, const.ClientType.agent],
)
def get_resource(
    tid: uuid.UUID, id: str, logs: bool = None, status: bool = None, log_action: const.ResourceAction = None, log_limit: int = 0
):
```

I mean having the Id in the path

Reply:
The advantage of the single `UnmanagedResource` parameter is that the protocol layer will do the type validation, while now it has to be done within the body of the API endpoint.

Reply:
Ah. Now I see. It's because the `unmanaged_resource_id` is part of the path of the URL. It makes sense now.